src/injecting/injections.rs

```diff
@@ -107,3 +107,110 @@ fn inject_heap_overflow(module: &mut Module) -> Result<(), String> {
+        let mut module = load_module_from_wasm(d.as_path()).unwrap();
+
+        let injection = Injection::InfiniteLoop;
+        assert!(injection.inject(&mut module).is_ok());
+
+        let function_body = get_function_body(&mut module);
+
+        let expected = vec![
+            Instruction::Loop(BlockType::NoResult),
+            Instruction::Nop,
+            Instruction::Br(0),
+            Instruction::End,
+        ];
+        assert!(function_body.code_mut().elements().starts_with(&expected))
+    }
+
+    #[test]
+    fn test_inject_jibberish_return_value() {
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("test.wasm");
+        let mut module = load_module_from_wasm(d.as_path()).unwrap();
+        let injection = Injection::JibberishReturnValue;
+        assert!(injection.inject(&mut module).is_ok());
+
+        let function_body = get_function_body(&mut module);
+
+        let expected = vec![Instruction::I64Const(123456789), Instruction::End];
+        assert!(function_body.code_mut().elements().starts_with(&expected))
+    }
+
+    #[test]
+    fn test_inject_stack_overflow() {
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("test.wasm");
+        let mut module = load_module_from_wasm(d.as_path()).unwrap();
+
+        let injection = Injection::StackOverflow;
+        assert!(injection.inject(&mut module).is_ok());
+
+        let function_body = get_function_body(&mut module);
+
+        let expected = vec![
+            Instruction::GetLocal(2),
+            Instruction::I32Const(2147483647),
+            Instruction::I32Store(2, 81000000),
+        ];
+        assert!(function_body.code_mut().elements().starts_with(&expected))
+    }
+
+    #[test]
+    fn test_inject_noops() {
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("test.wasm");
+        let mut module = load_module_from_wasm(d.as_path()).unwrap();
+
+        let injection = Injection::Noops;
+        assert!(injection.inject(&mut module).is_ok());
+
+        let function_body = get_function_body(&mut module);
+
+        let expected = vec![Instruction::Nop; 50]; // 500_000_000 is the actual number of injected instructions
+        assert!(function_body.code_mut().elements().starts_with(&expected))
+    }
+
+    #[test]
+    fn test_inject_heap_overflow() {
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("test.wasm");
+        let mut module = load_module_from_wasm(d.as_path()).unwrap();
+
+        let injection = Injection::HeapOverflow;
+        assert!(injection.inject(&mut module).is_ok());
+
+        let index: u32 = module.get_malloc_index().unwrap() as u32;
+        let function_body = get_function_body(&mut module);
+
+        let expected = vec![[Instruction::I32Const(33_554_431), Instruction::Call(index)]; 8]
+            .into_iter()
+            .flatten()
+            .collect::<Vec<Instruction>>();
```

Comment:
Clever monadic usage, nice :smile:

You could also consider something like this, for readability:

```rust
let expected = vec![Instruction::I32Const(33_554_431), Instruction::Call(index)]
  .into_iter()
  .cycle()
  .take(16)
  .collect::<Vec<_>>();
```